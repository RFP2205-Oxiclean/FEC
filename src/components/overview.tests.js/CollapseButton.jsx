import React, { useState } from "react";

const CollapseButton = ({ setIsHiding, isHiding }) => {
  return (
    <button
      data-testid="collapseButton"
      className="collapse-button"
      onClick={() => {
        setIsHiding(!isHiding);
      }}>
      Collapse
    </button>
  );
};

export default CollapseButton;
