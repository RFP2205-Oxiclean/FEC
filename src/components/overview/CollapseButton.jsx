import React, { useState } from "react";

const CollapseButton = ({ setIsHiding, isHiding }) => {
  return (
    <button
      data-testid="collapseButton"
      className="small-interactive-collapse-button"
      onClick={() => {
        setIsHiding(!isHiding);
      }}>
      Collapse
    </button>
  );
};

export default CollapseButton;
