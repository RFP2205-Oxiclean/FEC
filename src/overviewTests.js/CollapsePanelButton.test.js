/** @jest-environment jsdom */

import React from "react";
import { createRoot } from "react-dom/client";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import CollapseButton from "components/overview/CollapseButton.jsx";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

const container = document.createElement("div");
const root = createRoot(container);
root.render(<div></div>);

afterEach(cleanup);

it("should test text is 'Collapse' ", () => {
  const { getByTestId } = render(<CollapseButton></CollapseButton>);
  expect(getByTestId("collapseButton")).toHaveTextContent("Collapse");
});

it("should test Collapse has an onClick event", () => {
  const closePanelSpy = jest.fn();

  const { getByTestId } = render(<CollapseButton setIsHiding={closePanelSpy}></CollapseButton>);

  fireEvent.click(getByTestId("collapseButton"));

  expect(closePanelSpy).toHaveBeenCalled();
});
