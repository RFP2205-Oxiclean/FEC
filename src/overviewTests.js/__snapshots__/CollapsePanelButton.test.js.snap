// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="collapse-button"
  data-testid="collapseButton"
  onClick={[Function]}
>
  Collapse
</button>
`;
